File: week2/p10.c
#include <stdio.h>

int countZeros(int N, int A[][1000]) {
    int count = 0;
    int i = 0, j = N - 1; // start from top-right
    while (i < N && j >= 0) {
        if (A[i][j] == 0) {
            count += (j + 1); // all elements in this row till j are zero
            i++; // move to next row
        } else {
            j--; // move left
        }
    }
    return count;
}

int main() {
    int N;
    scanf("%d", &N);
    int A[1000][1000];
    for (int i = 0; i < N; i++)
        for (int j = 0; j < N; j++)
            scanf("%d", &A[i][j]);
    printf("%d\n", countZeros(N, A));
    return 0;
}

File: week2/p10.py
def countZeros(A):
    N = len(A)
    count = 0
    i, j = 0, N - 1  # start from top-right
    while i < N and j >= 0:
        if A[i][j] == 0:
            count += (j + 1)  # all elements from col 0 to j are zero
            i += 1
        else:
            j -= 1
    return count

# Example usage
N = int(input())
A = [list(map(int, input().split())) for _ in range(N)]
print(countZeros(A))
